### PR TITLE
Compile libssl on macOS automatically for Homebrew, and then complain at runtime if it fails

### PR DESCRIPTION
**Pull request by chrisseaton**

### CODE REVIEW

File: src/main/java/org/truffleruby/language/loader/RequireNode.java [hunk 1]
```diff
@@ -130,10 +130,37 @@ protected boolean require(VirtualFrame frame, String feature,
...
+                    }
+
+                    final CallTarget callTarget;
+
+                    try {
+                        callTarget = featureLoader.parseSource(source);
+                    } catch (JavaException e) {
+                        if (e.getCause() instanceof RuntimeException
+                                && e.getCause().getCause() instanceof IOException
+                                && e.getCause().getCause().getCause() instanceof UnsatisfiedLinkError) {
```
Comment (eregon) [inline src/main/java/org/truffleruby/language/loader/RequireNode.java]: This is a very long chain and seems quite brittle. I think it is going to break if it is not tested.

We could already remove one level by using sth like https://github.com/graalvm/truffleruby/blob/8a75511278048cf9806138ac28812b06f2ceb437/src/main/java/org/truffleruby/core/mutex/MutexOperations.java#L81-L87 in https://github.com/graalvm/truffleruby/blob/8a75511278048cf9806138ac28812b06f2ceb437/src/main/java/org/truffleruby/language/loader/FeatureLoader.java#L225 and make that a static method `rethrow` on say ExceptionOperations.

Comment (eregon) [inline src/main/java/org/truffleruby/language/loader/RequireNode.java]: Another way to make this cleaner would be to have a helper method that looks through causes in a loop and look for a UnsatisfiedLinkError.

Comment (chrisseaton) [inline src/main/java/org/truffleruby/language/loader/RequireNode.java]: Yes this is a systematic problem with our polyglot API.

Comment (chrisseaton) [inline src/main/java/org/truffleruby/language/loader/RequireNode.java]: Fixed with a search routine.

**Pull request merged**